modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RatedSearchHit.java

```diff
@@ -0,0 +1,114 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.rankeval;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.internal.InternalSearchHit;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+
+public class RatedSearchHit implements Writeable, ToXContent {
+
+    private final SearchHit searchHit;
+    private final Optional<Integer> rating;
+
+    public RatedSearchHit(SearchHit searchHit, Optional<Integer> rating) {
+        this.searchHit = searchHit;
+        this.rating = rating;
+    }
+
+    public RatedSearchHit(StreamInput in) throws IOException {
+        this(InternalSearchHit.readSearchHit(in), in.readBoolean() == true ? Optional.of(in.readVInt()) : Optional.empty());
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        searchHit.writeTo(out);
+        out.writeBoolean(rating.isPresent());
+        if (rating.isPresent()) {
+            out.writeVInt(rating.get());
+        }
+    }
+
+    public SearchHit getSearchHit() {
+        return this.searchHit;
+    }
+
+    public Optional<Integer> getRating() {
+        return this.rating;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        builder.field("hit", (ToXContent) searchHit);
+        builder.field("rating", rating.orElse(null));
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public final boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        RatedSearchHit other = (RatedSearchHit) obj;
+        // NORELEASE this is a workaround because InternalSearchHit does not properly implement equals()/hashCode(), so we compare their
```

Comment:
Capturing what we discussed f2f: I don't know how to best fix this - I'm ok with putting this in with NORELEASE for now, but this definitely needs either an equals implementation that doesn't rely on serialisation or it should be moved to the test code.